server/bin/pbench-audit-server.py

```diff
@@ -0,0 +1,852 @@
+                        "\t* Wrong prefix file names found in /.prefix directory:\n"
+                    )
+                    cnt = self.output_format(f, controller, "wrong_prefixes", cnt)
+                if controller in self.things_to_check["status_indicators"]._dict:
+                    if (
+                        "prefix_dir"
+                        in self.things_to_check["status_indicators"]._dict[controller]
+                    ):
+                        f.write("\t* Prefix directory, .prefix, is not a directory!\n")
+        f.write(
+            "\nend-{}: archive hierarchy: {}\n\n".format(self.config.TS[4:], self.path)
+        )
+        return cnt
+
+    def output_format(self, f, controller, controller_val, cnt):
+        f.write("\t  ++++++++++\n")
+        for val in sorted(self.things_to_check[controller_val]._dict[controller]):
+            f.write("\t  {}\n".format(val))
+        f.write("\t  ----------\n")
+        cnt += 1
+        return cnt
+
+
+def verify_subdirs(hier, controller, directories):
+    linkdirs = sorted(hier.config.LINKDIRS.split(" "))
+    if directories:
+        for dirent in directories:
+            if all(
+                [
+                    os.path.exists(os.path.join(hier.path, controller, dirent)),
+                    dirent != "_QUARANTINED",
+                    not dirent.startswith("WONT-INDEX"),
+                ]
+            ):
+                if dirent not in linkdirs:
+                    hier.add_unexpected_dirs(controller, dirent)
+    else:
+        hier.set_status(controller, "subdirs")
+
+    return 0
+
+
+def verify_prefixes(hier, controller):
+    prefix_dir = os.path.join(hier.path, controller, ".prefix")
+    if not os.path.exists(prefix_dir):
+        return
+    if not os.path.isdir(prefix_dir):
+        hier.set_status(controller, "prefix_dir")
+        return
+
+    prefixes = glob.iglob(os.path.join(prefix_dir, "*"))
+
+    for prefix in prefixes:
+        base_prefix = os.path.basename(prefix)
+        if not base_prefix.startswith("prefix.") and not base_prefix.endswith(
+            ".prefix"
+        ):
+            hier.add_non_prefixes(controller, base_prefix)
+        elif base_prefix.startswith("prefix."):
+            hier.add_wrong_prefixes(controller, base_prefix)
+
+    return
+
+
+def verify_tarball_names(hier, controller):
+    pass
+
+
+def verify_archive(hier):
+    controllers = glob.iglob(os.path.join(hier.path, "*"))
+
+    for controller in controllers:
+        if os.path.isdir(controller):
+            hier.add_controller(os.path.basename(controller))
+        else:
+            hier.add_bad_controller(os.path.basename(controller))
+
+    # now check each "good" controller and get the tarballs it contains
+    for controller in hier.controllers:
+        direct_entries = glob.iglob(os.path.join(hier.path, controller, "*"))
+        hidden_entries = glob.glob(os.path.join(hier.path, controller, ".*"))
+        controller_subdir = list()
```

Comment:
Move this lien down to just before the loop: that will keep all the `hidden_entries` stuff together.

Reply:
ok sure